### PR TITLE
Support relative paths in markdown images

### PR DESCRIPTION
**Pull request by runofthemillgeek**
Adds code from Hugo's built-in render-image hook and modifies it slightly to support relative image paths in markdown files.

Previously, we'd have to spell the absolute path from root of the hugo site or make directories for each individual post which is what Hugo recommend. But this was very inefficient. This change allows us to use editors like Obsidian, VSCode and have them render the images correctly. It also supports Hugo's page bundles for rendering images as a result. Based on the order, I believe page bundles are given priority so that works out.

i.e, if you have the following tree:

```
content
└── posts
   ├── _index.md
   ├── hello.md
   ├── images
   │  └── test.png
   └── test.png
```

You can now include images in `hello.md` in the following ways:

```
![](test.png)
![](images/test.png)
![](./test.png)
![](./images/test.png)
```

### CODE REVIEW

File: layouts/_default/_markup/render-image.html [hunk 1]
```diff
@@ -1,5 +1,33 @@
+{{/*
+To get page resources and relative paths to work. Copied as is from default
+hook:
+https://github.com/gohugoio/hugo/blob/89bd02/tpl/tplimpl/embedded/templates/_default/_markup/render-image.html
+*/}}
+{{- $u := urls.Parse .Destination -}}
+{{- $url := $u.String -}}
+{{- if not $u.IsAbs -}}
+  {{- $path := strings.TrimPrefix "./" $u.Path -}}
+  {{- $imgResource := .Page.Scratch.Get "typoNilVariable" -}}
```
Comment (runofthemillgeek) [inline layouts/_default/_markup/render-image.html:10]: This is intentional. Couldn't find a way to set `nil` inside templates.

**Comment by runofthemillgeek** [comment]
@tomfran was this accidentally closed when you force-pushed? 🤔

**Comment by tomfran** [comment]
Probably, sorry for that

**Comment by runofthemillgeek** [comment]
Huge reduction! :) I'll cherry-pick the commit and force-push, should be enough.

**Comment by runofthemillgeek** [comment]
Hmm okay, maybe need to open another.

**Pull request closed without merging**